# Determination - Deterministic rendering environment for music and art
# Copyright (C) 2024 Liu Hao <whiteaxe@tuta.io>
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, version 3.

# This patch gets rid of the large percentage of Ardour's code that forms its
# GUI, since we're only going to need to run Ardour's Lua environment and not
# the GUI.

--- a/wscript
+++ b/wscript
@@ -299,43 +299,43 @@ out = 'build'
 
 children = [
         # patched 3rd party libs
-        'libs/clearlooks-newer',
+        #'libs/clearlooks-newer',
         'libs/zita-resampler',
         'libs/zita-convolver',
         # optionally external libraries
         'libs/fluidsynth',
-        'libs/hidapi',
+        #'libs/hidapi',
         'libs/libltc',
         'libs/ptformat',
         'libs/qm-dsp',
         'libs/vamp-plugins',
         'libs/vamp-pyin',
-        'libs/aaf',
+        #'libs/aaf',
         # ytk/
-        'libs/tk/ztk',
-        'libs/tk/ydk-pixbuf',
-        'libs/tk/ydk',
-        'libs/tk/ytk',
-        'libs/tk/ztkmm',
-        'libs/tk/ydkmm',
-        'libs/tk/ytkmm',
-        'libs/tk/suil',
+        #'libs/tk/ztk',
+        #'libs/tk/ydk-pixbuf',
+        #'libs/tk/ydk',
+        #'libs/tk/ytk',
+        #'libs/tk/ztkmm',
+        #'libs/tk/ydkmm',
+        #'libs/tk/ytkmm',
+        #'libs/tk/suil',
         # core ardour libraries
         'libs/pbd',
         'libs/lua',
         'libs/midi++2',
         'libs/evoral',
         'libs/ctrl-interface',
-        'libs/surfaces',
+        #'libs/surfaces',
         'libs/panners',
         'libs/backends',
         'libs/temporal',
         'libs/ardour',
-        'libs/gtkmm2ext',
+        #'libs/gtkmm2ext',
         'libs/audiographer',
-        'libs/canvas',
-        'libs/widgets',
-        'libs/waveview',
+        #'libs/canvas',
+        #'libs/widgets',
+        #'libs/waveview',
         # plugins
         'libs/plugins/reasonablesynth.lv2',
         'libs/plugins/a-comp.lv2',
@@ -356,10 +356,10 @@ children = [
         'share/scripts',
         'share/web_surfaces',
         # frontends
-        'gtk2_ardour',
-        'headless',
+        #'gtk2_ardour',
+        #'headless',
         'luasession',
-        'session_utils',
+        #'session_utils',
         # shared helper binaries (plugin-scanner, exec-wrapper)
         'libs/fst',
         'libs/vfork',
@@ -368,10 +368,10 @@ children = [
 ]
 
 i18n_children = [
-        'gtk2_ardour',
+        #'gtk2_ardour',
         'libs/ardour',
-        'libs/gtkmm2ext',
-        'libs/tk/ytk',
+        #'libs/gtkmm2ext',
+        #'libs/tk/ytk',
 ]
 
 def set_compiler_flags (conf,opt):
